src/categorical_algebra/CSets.jl

```diff
@@ -268,9 +268,10 @@ Migrates an instance of one ACSet
 given dictionaries representing a functor between theories
 """
 
-function migrate!(Y::ACSet{CD}, X::ACSet, FOb::Dict, FHom::Dict) where {CD}
+function migrate!(Y::ACSet{CD, AD}, X::ACSet, FOb::Dict, FHom::Dict) where {CD, AD}
   @assert CD.ob ⊆ keys(FOb)
   @assert CD.hom ⊆ keys(FHom)
+  @assert AD.attr ⊆ keys(FHom)
```

Comment:
A julia idiom for these kinds of input validation checks is to use `condition || error("Error message about $var")`. The condition is always tested and since `||` is short circuiting, the error function is only called if the condition is false. I tend to use this `condition || error` style for validating inputs like this and `@assert` for program invariants which should only be false if there is a bug in the code.

Reply:
Thanks! I'll update that.